### PR TITLE
Updated frameworks documentation pages

### PR DESCRIPTION
**Pull request by zhenghh04**
Main changes: 
* Updated pytorch and tensorflow pages on Polaris and Aurora
* Added NCCL page on Polaris

### CODE REVIEW

**Comment by felker** [comment]
Could you give me write permissions to your fork? I would like to make some additional changes on this PR

- "OpenCL" https://github.com/argonne-lcf/user-guides/blob/70a37b69045f7078f73ded6548eddc92a410c6fa/mkdocs.yml#L211
- "oneCCL" https://github.com/argonne-lcf/user-guides/blob/70a37b69045f7078f73ded6548eddc92a410c6fa/mkdocs.yml#L205
- Fix link formatting in https://docs.alcf.anl.gov/aurora/data-science/frameworks/oneCCL/

File: docs/aurora/data-science/frameworks/pytorch.md [hunk 1]
```diff
@@ -138,13 +142,49 @@ export FI_LOG_LEVEL=warn
...
+export CCL_KVS_MODE=mpi
+export CCL_CONFIGURATION_PATH=""
+export CCL_CONFIGURATION=cpu_gpu_dpcpp
+export CCL_KVS_CONNECTION_TIMEOUT=600 
+
+export CCL_ZE_CACHE_OPEN_IPC_HANDLES_THRESHOLD=1024
+export CCL_KVS_USE_MPI_RANKS=1
+```
+
+Other optional setup for oneCCL.
```
Comment (felker) [inline docs/aurora/data-science/frameworks/pytorch.md:166]: could you explain what the purpose/advantage of the "optional setup"?

File: docs/aurora/data-science/frameworks/pytorch.md [hunk 2]
```diff
@@ -138,13 +142,49 @@ export FI_LOG_LEVEL=warn
...
+unset MPIR_CVAR_COLL_SELECTION_TUNING_JSON_FILE
+export INTELGT_AUTO_ATTACH_DISABLE=1
+export PALS_PING_PERIOD=240
+export PALS_RPC_TIMEOUT=240
+export MPIR_CVAR_GATHERV_INTER_SSEND_MIN_PROCS=-1 #to solve the sync send issue in Horovod seg fault
+export CCL_ATL_SYNC_COLL=1 #to avoid potential hang at large scale
+export CCL_OP_SYNC=1 #to avoid potential hang at large scale
 ```
 
+These setup will probably be included in the framework module file in future. But for now, users need to explicitly set these in the submission script.
```
Comment (felker) [inline docs/aurora/data-science/frameworks/pytorch.md:186]:
```suggestion
These environment variable settings will probably be included in the framework module file in the future. But for now, users need to explicitly set these in the submission script. 
```

File: docs/aurora/data-science/frameworks/tensorflow.md [hunk 3]
```diff
@@ -213,9 +215,66 @@ export MPIR_CVAR_ENABLE_GPU=0
 # This is to disable certain GPU optimizations like the use of XeLinks between
 # GPUs, collectives with GPU-placed data etc., in order to reduce `MPI_Init`
 # overheads. Benefits are application dependent.
+```
```
Comment (felker) [inline docs/aurora/data-science/frameworks/tensorflow.md:218]: Assuming the next ~100 lines are identical to those in `docs/aurora/data-science/frameworks/pytorch.md`, can you remove these and replace it with `pymdownx.snippets` section reference, as demonstrated in https://github.com/argonne-lcf/user-guides/pull/492#pullrequestreview-2406026019

File: docs/aurora/data-science/frameworks/tensorflow.md [hunk 4]
```diff
@@ -338,6 +396,36 @@ export NUMEXPR_NUM_THREADS=64
 # or equal to '64' or to increase the 'NUMEXPR_MAX_THREADS' to the available
 # number of threads. Both of these variables can be set manually.
 
+
+## CCL setup
```
Comment (felker) [inline docs/aurora/data-science/frameworks/tensorflow.md:400]: same comment about `pymdownx.snippets` as earlier

File: docs/polaris/applications-and-libraries/libraries/nccl.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# NCCL 
+
+NCCL (pronounced "Nickel") is a stand-alone library of standard communication routines for GPUs, implementing all-reduce, all-gather, reduce, broadcast, reduce-scatter, as well as any send/receive based communication pattern. It has been optimized to achieve high bandwidth on platforms using PCIe, NVLink, NVswitch, as well as networking using InfiniBand Verbs or TCP/IP sockets. NCCL supports an arbitrary number of GPUs installed in a single node or across multiple nodes, and can be used in either single- or multi-process (e.g., MPI) applications.
```
Comment (felker) [inline docs/polaris/applications-and-libraries/libraries/nccl.md:3]:
```suggestion
NVIDIA NCCL (pronounced "Nickel") is a standalone library of standard communication routines for GPUs, implementing all-reduce, all-gather, reduce, broadcast, reduce-scatter, as well as any send/receive based communication pattern. It has been optimized to achieve high bandwidth on platforms using PCIe, NVLink, NVswitch, as well as networking using InfiniBand Verbs or TCP/IP sockets. NCCL supports an arbitrary number of GPUs installed in a single node or across multiple nodes, and can be used in either single- or multi-process (e.g., MPI) applications.
```

**Review by felker — CHANGES_REQUESTED** [review]

File: docs/polaris/applications-and-libraries/libraries/nccl.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+NCCL is a key library for scaling AI applications on Nvidia system. The conda module on Polaris are built with NCCL as the communication backend for distributed training. But HPC applications can also chose NCCL for communication over MPI. The library is available in the following folder: ```/soft/libraries/nccl```. 
+
+We have done extensive performance tests and identified the following best environment setup.
```
[the remark this answers is not in the document]
Comment (zhenghh04) [reply docs/polaris/applications-and-libraries/libraries/nccl.md:7]: Added a couple sentences for this.

File: docs/aurora/data-science/frameworks/pytorch.md (hunk 1 above)
Comment (zhenghh04) [reply docs/aurora/data-science/frameworks/pytorch.md]: added description in the oneCCL page.

**Comment by zhenghh04** [comment]
@felker I addressed all the comments you put. Please review it again.

File: docs/polaris/applications-and-libraries/libraries/nccl.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+export FI_CXI_DEFAULT_CQ_SIZE=131072
+```
+This setup will lead to 2-3x performance improvement. For details, please refer to: https://github.com/argonne-lcf/alcf-nccl-tests. 
+
+As of now (October 29, 2029), these setup has been included in the conda module on Polaris as one can confirm as follows: 
+```bash 
+module load conda
+env | grep NCCL
+env | grep FI
+```
```
[the remark this answers is not in the document]
Comment (felker) [reply docs/polaris/applications-and-libraries/libraries/nccl.md:30]: FYI I was referring to https://squidfunk.github.io/mkdocs-material/reference/admonitions/

File: docs/polaris/applications-and-libraries/libraries/nccl.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+**WARNING**
+For some applications such as Megatron-DeepSpeed, enabling AWS plugin will cause hang or NCCL timeout issue. If so, please disable it by 
+
+```bash
+unset NCCL_NET_GDR_LEVEL NCCL_CROSS_NIC NCCL_COLLNET_ENABLE NCCL_NET
+```
```
Comment (felker) [inline docs/polaris/applications-and-libraries/libraries/nccl.md:37]:
```suggestion
!!! warning
    For some applications such as Megatron-DeepSpeed, enabling AWS plugin will cause hang or NCCL timeout issue. If so, please disable it by:
    ```bash
    unset NCCL_NET_GDR_LEVEL NCCL_CROSS_NIC NCCL_COLLNET_ENABLE NCCL_NET
    ```
```

**Pull request merged**